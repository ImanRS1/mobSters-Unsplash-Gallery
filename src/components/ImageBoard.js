import React from 'react';
import ImageCard from './ImageCard';
import styled from "styled-components";

const ImageBoard = props => {

  const imageGenerator = data => {
    return data.map(img => <ImageCard url={img.urls.small} author={img.user.name} descript={img.alt_description} key={img.id} />)
  }

  return (
    <ImageContainer>
      {imageGenerator(props.data)}
    </ImageContainer>
  )
}

const ImageContainer = styled.section`
  max-width: 1440px;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
`;

export default ImageBoard
