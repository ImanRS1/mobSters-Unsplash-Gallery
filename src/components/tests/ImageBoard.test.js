import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import ImageBoard from '../ImageBoard';
import renderer from 'react-test-renderer';

const data = [
  {
    "urls": {"small": "https://picsum.photos/id/237/200/300"},
    "user": {"name": "Mariana"},
    "alt_description": "Hello",
    "id": "1"
  },
  {
    "urls": {"small": "https://picsum.photos/id/237/200/300"},
    "user": {"name": "Mariana"},
    "alt_description": "Hello",
    "id": "2"
  } 
  ];

describe('Basic ImageBoard test', () => {
  const div = document.createElement('div');

  test('Renders', () => {
    ReactDOM.render(<BrowserRouter><ImageBoard data={data} /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  test('Images render correctly', () => {
    const imageBoardComponent = renderer.create(<BrowserRouter><ImageBoard data={data} /></BrowserRouter>).toJSON();
    expect(imageBoardComponent).toMatchSnapshot();
  })
  test('Correct number of rendered images', () => {
    ReactDOM.render(<BrowserRouter><ImageBoard data={data} /></BrowserRouter>, div);
    expect(div.children[0].children.length).toBe(data.length);
  });
});
