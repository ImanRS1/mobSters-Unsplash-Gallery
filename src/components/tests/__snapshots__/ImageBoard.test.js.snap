// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic ImageBoard test Images render correctly 1`] = `
<section
  className="sc-jRQBWg jIzGGa"
>
  <article
    className="sc-gsDKAQ dRRDgd"
  >
    <div
      className="sc-bdvvtL gHIAtL"
    >
      <div
        className="sc-dkPtRN ljzhnS"
      >
        <img
          alt="Hello"
          className="sc-eCImPb exxfQC"
          src="https://picsum.photos/id/237/200/300"
        />
      </div>
      <div
        className="sc-hKwDye ghZbUD"
      >
        <h3>
          Mariana
        </h3>
        <p>
          Hello
        </p>
      </div>
    </div>
  </article>
  <article
    className="sc-gsDKAQ dRRDgd"
  >
    <div
      className="sc-bdvvtL gHIAtL"
    >
      <div
        className="sc-dkPtRN ljzhnS"
      >
        <img
          alt="Hello"
          className="sc-eCImPb exxfQC"
          src="https://picsum.photos/id/237/200/300"
        />
      </div>
      <div
        className="sc-hKwDye ghZbUD"
      >
        <h3>
          Mariana
        </h3>
        <p>
          Hello
        </p>
      </div>
    </div>
  </article>
</section>
`;
